src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -158,7 +167,7 @@ func (r *StatefulSetResource) Obj() (k8sclient.Object, error) {
 					},
 					InitContainers: []corev1.Container{
 						{
-							Name:		"redpanda-configurator",
+							Name:		configuratorContainerName,
```

Comment:
this is actually interesting. I know it's out of scope, but how do we plan to update the configurator? I think we should note that somewhere in the document. I assume we just need to try to update the configmap and that might be it. Also rolling update the pods so that they see the new config

Reply:
I think the `initContainer` as it is, should always have the same version as the application container.

It would probably be best for the operator to create the `configMap` in the `emptyDir` directly, and drop the `initContainer`. 

Almost all of the configuration can be moved to rpk redpanda start args, except node-id and seeds - an example is in the [helm-charts](https://github.com/vectorizedio/helm-charts/pull/25/commits/e38731b257984d4228c49eb2af2785048d6298e7#diff-f74d1889126f524c4b7e7e7836a376c58b0ca5aabe3720e03732b0e25d6ee380R71)

Given that Redpanda configuration is having a massive overhaul, it might not be worth the effort for now.

Reply:
> oh we use configmap to also bundle the shell script - I was mainly talking about that. Sorry I should have made it more clear https://github.com/vectorizedio/redpanda/blob/dev/src/go/k8s/pkg/resources/configmap.go#L39

Ahh, I see. In my previous company we developed [Wave](https://github.com/wave-k8s/wave) to automagically kick pods based on changes to config - I think it supports statefulSets although I never used it for them, but the gist of it is the same.

Reply:
Init container is updated along the main container.